app/shared/models/github_provider_credential.rb

```diff
@@ -30,21 +30,25 @@ def initialize(id: nil, email: nil, full_name: nil, api_token: nil)
       @id = id || SecureRandom.uuid
       @email = email
       @full_name = full_name
-      @api_token = api_token
```

Comment:
this used to be `self.api_token=` which called the setter we wrote, so when `@api_token` was swapped, it no longer provided access to the token since we overrode the getter and setter to decrypt and encrypt `@encrypted_api_token` on the fly.

Reply:
Ahh, my apologies, that makes sense - was an overlook on my part. Thank you for finding the issue and fixing it!

Reply:
No worries! It's a tricky one, but really good to learn from!